File: tech-studio-projects/financial-industry/crypto-trading-simulation/java/src/network/heartbeat/HeartbeatManager.cpp
#include "HeartbeatManager.h"
#include <iostream>
#include <chrono>

HeartbeatManager::HeartbeatManager() : running(false) {}

void HeartbeatManager::start() {
    running = true;
    std::thread(&HeartbeatManager::sendHeartbeats, this).detach();
}

void HeartbeatManager::stop() {
    running = false;
}

void HeartbeatManager::sendHeartbeats() {
    while (running) {
        std::this_thread::sleep_for(std::chrono::seconds(5));
        std::cout << "Sending heartbeat..." << std::endl; // Placeholder for actual heartbeat logic
    }
}

File: tech-studio-projects/financial-industry/crypto-trading-simulation/java/src/network/heartbeat/HeartbeatManager.java
// HeartbeatManager.java
public class HeartbeatManager {

    private volatile boolean running; // Use volatile for thread safety

    // Constructor
    public HeartbeatManager() {
        this.running = false;
    }

    // Method to start sending heartbeats
    public void start() {
        running = true;
        new Thread(this::sendHeartbeats).start();
    }

    // Method to stop sending heartbeats
    public void stop() {
        running = false;
    }

    // Method that runs in a separate thread to send heartbeats
    private void sendHeartbeats() {
        while (running) {
            try {
                Thread.sleep(5000); // Sleep for 5 seconds
                System.out.println("Sending heartbeat..."); // Placeholder for actual heartbeat logic
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt(); // Restore interrupt status
                System.out.println("Heartbeat thread interrupted");
            }
        }
    }
}
